R/read_from_servers.R

```diff
@@ -0,0 +1,210 @@
+#'   )
+#'
+#'   # import data where query parameters is within an SQL query
+#'   authors_list <- read_server(
+#'     conn  = conn,
+#'     query = "select * from author"
+#'   )
+#' }
+#' @export
+read_server <- function(conn, query) {
+  stopifnot("Invalid connection object!" = inherits(conn, "Pool"))
+  if (!checkmate::test_character(query, len = 1L, null.ok = FALSE)) {
+    checkmate::assert_list(query, min.len = 1L)
+    checkmate::assert_vector(query[["select"]], min.len = 0L, null.ok = TRUE,
+                             any.missing = FALSE)
+    if (!checkmate::test_character(query[["filter"]], null.ok = TRUE,
+                                   any.missing = FALSE)) {
+      checkmate::assert_vector(query[["filter"]], null.ok = TRUE,
+                                any.missing = FALSE)
+    }
+    checkmate::assert_character(query[["table"]], len = 1L, null.ok = FALSE,
+                                any.missing = FALSE)
+  }
+
+  # When the query parameter is a list, the below function will be used
+  # construct the SQL query to be used to fetch the data from the specified
+  # table
+  if (is.list(query)) {
+    query <- server_make_query(
+      table_name = query[["table"]],
+      connexion  = conn,
+      filter     = query[["filter"]],
+      select     = query[["select"]]
+    )
+  }
+
+  # fetch the data using the SQL query
+  final_data <- server_fetch_data(connexion = conn, query = query)
+
+  # return the datasets of interest
+  return(final_data)
+}
+
+#' Make SQL query from the list of query parameters
+#'
+#' @param table_name The name of the table in the server
+#' @param connexion A connection object created from the `authenticate()`
+#'    function
+#' @param filter A vector or a comma-separated string of subset of subject IDs
+#' @param select A vector of column names
+#'
+#' @return A string with the constructed SQL query from the provided query
+#'    parameter.
+#' @examples
+#' # establish the connection
+#' \donttest{
+#'   # establish the connection to the database
+#'   conn <- authenticate(
+#'     from          = "mysql-rfam-public.ebi.ac.uk",
+#'     type          = "MySQL",
+#'     user_name     = "rfamro",
+#'     password      = "",
+#'     driver_name   = "",
+#'     db_name       = "Rfam",
+#'     port          = 4497
+#'   )
+#' }
+#' result <- server_make_query(
+#'   table_name = "author",
+#'   connexion  = conn,
+#'   filter     = filter,
+#'   select     = select
+#' )
+#' @keywords internal
+#'
+server_make_query <- function(table_name,
+                              connexion,
+                              filter,
+                              select) {
+  if (is.null(filter) && is.null(select)) {
+    # subset all rows and all columns
+    query          <- sprintf("select * from %s", table_name)
+  } else if (!is.null(select) && is.null(filter)) {
+    # subset all rows and specified columns
+    target_columns <- paste(select, collapse = ", ")
```

Comment:
```suggestion
    target_columns <- glue::glue_collapse(select, sep = ", ")
```

Reply:
Thanks @Degoot-AM - I will update the figure accordingly.

`paste()` is no longer spotted out by lintr. I think it is better to use `paste()` and not `glue_collapse()` to avoid taking {glue} as a dependency.